grow/translators/google_sheets.py

```diff
@@ -292,39 +296,147 @@ def _create_sheets_request(self, catalogs, source_lang, spreadsheet_id,
-            existing_values.pop(0)  # Remove header row.
+            for x in range(self.HEADER_ROW_COUNT):
+                existing_values.pop(0)  # Remove header rows.
             for value in existing_values:
-                source = value[0]
-                translation = value[1] if len(value) > 1 else None
                 if value not in catalog:
+                    source = value[0]
+                    translation = value[1] if len(value) > 1 else None
                     catalog.add(source, translation, auto_comments=[],
                                 context=None, flags=[])
-            sheet = self._create_sheet_from_catalog(catalog, source_lang,
-                    include_obsolete=include_obsolete)
-            sheet['properties']['sheetId'] = sheet_id
-            requests.append({
-                'updateSheetProperties': {
-                    'fields': '*',
-                    'properties': sheet['properties'],
-                },
-            })
+
+            # Perform a diff of the existing data to what the catalog provides
+            # to make targeted changes to the spreadsheet and preserve meta
+            # information--such as comments.
+            existing_rows, new_rows, removed_rows = self._diff_data(
+                    existing_values, catalog)
+
+            # Update the existing values in place.
+            if len(existing_rows):
+                row_data = []
+                for value in existing_rows:
+                    row_data.append(self._create_catalog_row(
+                            value['source'], value['translation'],
+                            value['locations']))
+                # NOTE This is not (yet) smart enough to only update small sections
+                # with the updated information. Hint: Use value['updated'].
+                requests.append({
+                    'updateCells': {
+                        'fields': '*',
+                        'start': {
+                            'sheetId': sheet_id,
+                            'rowIndex': self.HEADER_ROW_COUNT, # Skip header row.
+                            'columnIndex': 0,
+                        },
+                        'rows': row_data,
+                    },
+                })
+
+            # Append new values to end of sheet.
+            if len(new_rows):
+                row_data = []
+                for value in new_rows:
+                    row_data.append(self._create_catalog_row(
+                            value['source'], value['translation'],
+                            value['locations']))
+
+                requests.append({
+                    'appendCells': {
+                        'sheetId': sheet_id,
+                        'fields': '*',
+                        'rows': row_data,
+                    },
+                })
+
+            # Remove obsolete rows if not included.
+            if not include_obsolete and len(removed_rows):
```

Comment:
Alright, I've updated it to make it a `prune_obsolete` and removed the `include_obsolete` from the upload command. It will now just use the config version of the setting for doing the extraction.

Also made it so that the option to prune is off by default.